strawberry/extensions/query_depth_limiter.py

```diff
@@ -28,27 +28,57 @@
 from __future__ import annotations
 
 import re
-from typing import TYPE_CHECKING, Callable, Dict, Iterable, List, Optional, Type, Union
+import warnings
+from dataclasses import dataclass
+from typing import (
+    Callable,
+    Dict,
+    Iterable,
+    List,
+    Optional,
+    Type,
+    Union,
+)
 
 from graphql import GraphQLError
 from graphql.language import (
+    BooleanValueNode,
+    DefinitionNode,
     FieldNode,
+    FloatValueNode,
     FragmentDefinitionNode,
     FragmentSpreadNode,
     InlineFragmentNode,
+    IntValueNode,
+    ListValueNode,
+    Node,
+    ObjectValueNode,
     OperationDefinitionNode,
+    StringValueNode,
+    ValueNode,
 )
-from graphql.validation import ValidationRule
+from graphql.validation import ValidationContext, ValidationRule
 
 from strawberry.extensions import AddValidationRules
 from strawberry.extensions.utils import is_introspection_key
 
-if TYPE_CHECKING:
-    from graphql.language import DefinitionNode, Node
-    from graphql.validation import ValidationContext
+IgnoreType = Union[Callable[[str], bool], re.Pattern, str]
+
+FieldArgumentType = Union[
+    bool, int, float, str, List["FieldArgumentType"], Dict[str, "FieldArgumentType"]
+]
+FieldArgumentsType = Dict[str, FieldArgumentType]
```

Comment:
we might want to move these out at some point, or reuse something classes we already have